src/bootstrap/src/ferrocene/doc.rs

```diff
@@ -812,7 +812,6 @@ impl Step for EmptyStep {
 // Note: this function is correct for the use made in this module, but it will not work correctly
 // if paths don't exists or the paths do not have any segments in common.
 fn relative_path(base: &Path, path: &Path) -> PathBuf {
-    let base = std::fs::canonicalize(base).unwrap_or_else(|_| base.to_path_buf());
```

Comment:
We need to be super careful with this, have you checked it doesn't break anything?

`relative_path` has been added because if any full path gets into Sphinx signatures break.

Reply:
Will this step get run at all on Windows? I bumped into this while running `x.py doc` thinking I was on Linux if I recall. It may be I can just yank this.

It was causing an issue on Windows due to `.canonicalize()` producing UNC paths.